module-4/backend/week_four.md

```diff
@@ -5,18 +5,41 @@ Fork this respository. Answer the questions to the best of your ability. Try to
 Note: When you're done, submit a PR. 
 
 * What's the purpose of a project management tool?
+  Project management tools help us organize our tasks, especially when we work in teams.
+  
 * What are the stages of a card when using Pivotal Tracker?
+  Start, Finish, Deliver, Accept/Reject
+  
 * What are the four types of cards available in Pivotal Tracker?
+  Chore, Feature, Bug, Release
+  
 * What's your favorite project management tool?
+  So far, I like Pivotal the best. Waffle is too dependent on Github issues and doesn't sync well, and Trello is too loose.
+  
 * Why do we create staging environments?
+  Staging has all of the benefits of production, but the safety of not messing up production. We can test more thoroughly with staging.
+  
 * What are the characteristics of a good README (in your opinion)?
+  I think a good README should be concise, clear, and separated into appropriate categories (Getting Started, FAQs, etc).
+  
 * What's one main improvement you're going to make to your code regarding accessibility issues?
+  I'm going to test my ability to tab through the page and make certain content is available for screenreaders.
+  
 * What are some basic security concerns to be aware of when building applications?
+  SQL injection, fields in the strong params, not telling users which field is incorrect when they fail to login.
+  
 * Why is continuous integration helpful/important?
+  Continuous integration is more agile, and it promotes modularity in code that allows for smoother transitions. It also creates a smooth deploy process.
+  
 * What are some continuous integration tools?
-
+  Travis, Jenkins, Strider
 #### Review  
 
 * What are some characteristics of "good" git workflow?
+  Atomic commits, never pushing with failing tests, creating feature branches
+  
 * What are the four fundamental concepts of object oriented programming?
+  Encapsulation, Abstraction, Modularity, DRY
```

Comment:
encapsulation, abstraction, polymorphism, and inheritance are the common four fundamentals of OOP 👍